src/cargo-fmt/main.rs

```diff
@@ -284,6 +284,15 @@ impl Target {
             edition: target.edition.clone(),
         }
     }
+
+    /// `Cargo.toml` file to format.
+    pub fn manifest_target(manifest_path: PathBuf) -> Self {
+        Target {
+            path: manifest_path,
+            kind: String::from("manifest"),
+            edition: String::from("2021"), // The value doesn't matter for formatting Cargo.toml.
+        }
+    }
```

Comment:
hmmm 🤔 I wasn't expecting to see changes in `src/cargo-fmt`. I think I understand why these changes and other `src/cargo-fmt` changes were needed otherwise `cargo fmt` wouldn't format the `Cargo.toml` files. Do I have that correct?

Reply:
Yes. Previously `cargo fmt` only runs `rustfmt` on lib & bin files.

Reply:
Personally I feel like we should add the `cargo-fmt` support in a follow up PR. The way things are currently implemented `cargp-fmt` will always invoke `rustfmt path/to/Cargo.toml` even if we haven't set `format_cargo_toml` in our `rustfmt.toml`. Probably worth discussing whether or not `Cargo.toml` formatting in `cargo fmt` should be gated by a command line option like `cargo fmt --format-cargo-toml` that will then invoke rustfmt like `rustfmt --config=format_cargo_toml=true path/to/Cargo.toml`.

Reply:
Since this PR is already on hold for long, I think we can directly discuss and resolve concerns here without splitting it. 😄 Otherwise this feature won’t be too useful for users or won’t be merged soon as before. But if you think the remaining is ready to merge but this part needs major improvements. I’m also OK with that.

As for the feature gate problem, I think it makes some sense to let cargo fmt to control it. But if we want to do so, the rustfmt config would become unnecessary. 

Comparing the two way, I feel using rustfmt config is more user-friendly and more consistent. IMO this feature is more like toggling the rustfmt’s behavior than controlling the scope to be formatted. Cargo.toml is in the same logical unit of a package. With the same reason, I think it is reasonable for cargo fmt to always call rustfmt on Cargo.toml. Alternatively, maybe we can also let rustfmt finds the Cargo.toml or sth. I mean logically they are the same, and current implementation is a quick and simple way to make it work.

Reply:
Also cc @calebcartwright for suggestions about how we can move forward.